libdevcore/MemoryDB.cpp

```diff
@@ -33,30 +49,31 @@ void MemoryDBWriteBatch::kill(Slice _key)
 std::string MemoryDB::lookup(Slice _key) const
 {
     std::string value;
-    std::lock_guard<std::mutex> lock(m_mutex);
-    if (m_db.count(_key.data()))
+    dev::Guard lock(m_mutex);
+    auto const& it = m_db.find(_key.toString());
+    if (it != m_db.end())
     {
-        value = m_db.find(_key.data())->first;
+        value = it->first;
     }
     return value;
 }
 
 bool MemoryDB::exists(Slice _key) const
 {
-    std::lock_guard<std::mutex> lock(m_mutex);
+    dev::Guard lock(m_mutex);
     return m_db.count(_key.data()) != 0;
```

Comment:
must be `_key.toString()`

Reply:
Also in `MemoryDBWriteBatch` methods above

Reply:
Sorry about that, I thought I had gotten all of them.